client/js/compute.js

```diff
@@ -0,0 +1,554 @@
+const compute_ops = [
+    {
+      name: '=',
+      desc: 'sets the variable to the value of x',
+      sample: 'var a = ${x}',
+      call: function(v, x) { return v = v = x }
+    },
+    {
+      name: '+',
+      desc: 'returns the sum of two numbers or add/join/concat two strings together',
+      sample: 'var a = ${x} + ${y}',
+      call: function(v, x, y) { return v = x + y }
+    },
+    {
+      name: '-',
+      desc: 'returns the difference of two numbers (subtract y from x)',
+      sample: 'var a = ${x} - ${y}',
+      call: function(v, x, y) { return v = x - y }
+    },
+    {
+      name: '*',
+      desc: 'returns the product of two numbers (multiply x and y)',
+      sample: 'var a = ${x} * ${y}',
+      call: function(v, x, y) { return v = x * y }
+    },
+    {
+      name: '**',
+      desc: 'returns the result of raising x to the power of y (exponent)',
+      sample: 'var a = ${x} ** ${y}',
+      call: function(v, x, y) { return v = x ** y }
+    },
+    {
+      name: '/',
+      desc: 'returns the division of two numbers',
+      sample: 'var a = ${x} / ${y}',
+      call: function(v, x, y) { return v = x / y }
+    },
+    {
+      name: '%',
+      desc: 'returns the remainder/rest/modal of the division of two numbers',
+      sample: 'var a = ${x} % ${y}',
+      call: function(v, x, y) { return v = x % y }
+    },
+    {
+      name: '<',
+      desc: 'returns true if x is less than y',
+      sample: 'var a = ${x} < ${y}',
+      call: function(v, x, y) { return v = x < y }
+    },
+    {
+      name: '<=',
+      desc: 'returns true if x is less than or equal to y',
+      sample: 'var a = ${x} <= ${y}',
+      call: function(v, x, y) { return v = x <= y }
+    },
+    {
+      name: '==',
+      desc: 'returns true if x is equal to y',
+      sample: 'var a = ${x} == ${y}',
+      call: function(v, x, y) { return v = x == y }
+    },
+    {
+      name: '!=',
+      desc: 'returns true if x is not equal to y',
+      sample: 'var a = ${x} != ${y}',
+      call: function(v, x, y) { return v = x != y }
+    },
+    {
+      name: '>=',
+      desc: 'returns true if x is greater than or equal to y',
+      sample: 'var a = ${x} >= ${y}',
+      call: function(v, x, y) { return v = x >= y }
+    },
+    {
+      name: '>',
+      desc: 'returns true if x is greater than y',
+      sample: 'var a = ${x} > ${y}',
+      call: function(v, x, y) { return v = x > y }
+    },
+    {
+      name: '&& (logical AND)',
+      desc: 'returns true if both x and y are true or returns the value of x or y if they are not boolean values',
+      sample: 'var a = ${x} >= ${y}',
+      call: function(v, x, y) { return v = x && y }
+    },
+    {
+      name: '|| (logical OR)',
+      desc: 'returns true if x or y or both are true or returns the value of x or y if they are not boolean values',
+      sample: 'var a = ${x} >= ${y}',
+      call: function(v, x, y) { return v = x || y }
+    },
```

Comment:
I'm pretty sure line 83 is supposed to be:
sample: 'var a =${x} && ${y}',
and line 89 is supposed to be:
sample: 'var a =${x} || ${y}',